Sources/Common/Core/Math/index.js

```diff
@@ -798,21 +812,21 @@ export function jacobiN(a, n, w, v) {
 
     for (ip = 0; ip < n - 1; ip++) {
       for (iq = ip + 1; iq < n; iq++) {
-        g = 100.0 * Math.abs(a[ip][iq]);
+        g = 100.0 * Math.abs(a_temp[ip * n + iq]);
 
         // after 4 sweeps
         if (
           i > 3 &&
           Math.abs(w[ip]) + g === Math.abs(w[ip]) &&
           Math.abs(w[iq]) + g === Math.abs(w[iq])
         ) {
-          a[ip][iq] = 0.0;
-        } else if (Math.abs(a[ip][iq]) > tresh) {
+          a_temp[ip * n + iq] = 0.0;
```

Comment:
it no longer modifies input `a`, is it on purpose ?

Reply:
Yes, the modification of `a` during computation of eigen values and vectors made the test of diagonalize3x3 function tricky. The verification of `D = P^-1 A P` is done by computing the eigen vectors on one hand and computing the diagonalization on the another hand. As `a` was changed in the computation of the eigen values and vectors `D = P^-1 A P` was false even though the computation of `D` and `P` was correct. Since the modified value of `a` was not meaningfull, I thought it was better to copy the matrix inside of jacobiN before doing the computation than having to copy it whenever we want to pass it to jacobiN. It is not the same implementation as vtk C++ though.